test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2020 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctl
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/jetstack/cert-manager/cmd/ctl/pkg/create/certificaterequest"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+	"github.com/jetstack/cert-manager/test/integration/framework"
+)
+
+func TestCtlCreateCR(t *testing.T) {
+	config, stopFn := framework.RunControlPlane(t)
+	defer stopFn()
+
+	ctx, cancel := context.WithTimeout(context.TODO(), time.Second*20)
+	defer cancel()
+
+	// Build clients
+	_, _, cmCl, _ := framework.NewClients(t, config)
+
+	// Create tmp directory and cd into it to store private key files
+	dir, err := ioutil.TempDir(".", "tmp")
```

Comment:
This will not do what you expect, especially in combination with the cleanup method below.

```
func TempDir(dir, pattern string) (name string, err error)

TempDir creates a new temporary directory in the directory dir. The directory name is generated by taking pattern and applying a random string to the end. If pattern includes a "*", the random string replaces the last "*". TempDir returns the name of the new directory. If dir is the empty string, TempDir uses the default directory for temporary files (see os.TempDir). Multiple programs calling TempDir simultaneously will not choose the same directory. It is the caller's responsibility to remove the directory when no longer needed.
```

(from https://golang.org/pkg/io/ioutil/#TempDir)

Here you've specified the 'directory' as `.`, which will use the current working directory. The 'pattern' will then be `tmp`,which with some additional text generated would end out as something like `tmpXXXXXX` in the current directory.

The cleanup function right now looks like this:

```
func cleanUpTmpDir(dir string) error {
	if err := os.Chdir(".."); err != nil {
		return err
	}
	if err := os.RemoveAll(dir); err != nil {
		return err
	}
	return nil
}
```

We should _never_ be relying on the current working directory, in _any_ part of the test case, especially when deleting files.

Instead, I'd expect to see something like: `dir, err := ioutil.TempDir("", "tmp-ctl-test-*")` (notice the `*` character gets replaced with the random component as per the godoc).

The cleanup function I'd expect to simply call `os.RemoveAll(dir)`. Something like this also allows us to 'fail' the test if something funky goes on during cleanup:

```
func removeDir(t *testing.T, dir string) {
	if err := os.RemoveAll(dir); err != nil {
		t.Fatal(err)
	}
}
```

Notice that this function no longer returns an error (which was being ignored a few lines below as part of the `defer` anyway 😄)